category_theory/category.lean

```diff
@@ -32,22 +32,25 @@ powerful tactics.
 def_replacer obviously
 @[obviously] meta def obviously' := tactic.tidy
 
+class has_hom (obj : Type u) : Type (max u (v+1)) :=
```

Comment:
I'm stlil not wild about the name `has_hom` for this structure which also contains the other data of a category. I'll try to think of a better alternative.

Reply:
In some alternate proposals for the algebraic hierarchy, we used names like `group_struct` for the data without the laws. Would that work here?

Reply:
I like it